tests/component/component.cpp

```diff
@@ -499,3 +499,30 @@ TEST(Component, constructors)
     libcellml::ComponentPtr c3 = std::move(c2);
     EXPECT_EQ("my_name", c3->name());
 }
+
+TEST(Component, multiParentWithAddComponentBugIssue399)
```

Comment:
This test looks good to me and would give the behaviour I would expect.

Reply:
Sadly, it doesn't look good to me.  We should be able to add `child3` to `child2` and have `child3` as . a child of both `parent` and `child2`.

Reply:
Our difficulty will come if we ask `child3` who its parent is.

Reply:
Three options?
- allow more than one parent until validation, so the user gets the full picture
- update parent and child at the same time, preventing errors but also giving network dependent on the order in which items were added
- forbid changing the network if it will break like this, and alert the user